### PR TITLE
Added Painter's Partition Problem

### LINKED ISSUE #874: Painter's Partition Problem in C++, Java, Python
Please assign in to me.

### PR DESCRIPTION
**Pull request by MastersAbh**
Fixes #874  

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Sample Input / Output.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Changes proposed in this pull request:

-  Added Painter's Partition Problem in C++, Java and Python

#### Languages Used:
-  C++
- Java
- Python

#### Files Added:
- Painter's_Partition_Problem.cpp
- Painter's_Partition_Problem.java
- Painter's_Partition_Problem.py

### CODE REVIEW

File: Painter's_Partition_Problem/Painter's_Partition_Problem.cpp [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+/*
+    This function returns the minimum no. of painters required to
+    paint the board given that a painter can paint consecutive mid
+    units of board at max.
+*/
+lint painters(lint board[], lint n, lint mid) {
+    lint painter = 0;
+    lint time = 0;
+    for(lint i=0; i<n; i++) {
```
Comment (somya-kapoor) [inline Painter's_Partition_Problem/Painter's_Partition_Problem.cpp]: Please add space around =,<

File: Painter's_Partition_Problem/Painter's_Partition_Problem.cpp [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        }
+    }
+    return low;
+}
+
+int main(){
+    lint n; // Enter the number of boards.
+    cin>>n;
+    lint board[n];  // Length of boards.
+    for(lint i=0; i<n; i++){
```
Comment (somya-kapoor) [inline Painter's_Partition_Problem/Painter's_Partition_Problem.cpp]: Add space around =,<

File: Painter's_Partition_Problem/Painter's_Partition_Problem.cpp [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        cin>>board[i];
```
Comment (somya-kapoor) [inline Painter's_Partition_Problem/Painter's_Partition_Problem.cpp]: Add space around >>

File: Painter's_Partition_Problem/Painter's_Partition_Problem.cpp [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+int main(){
+    lint n; // Enter the number of boards.
+    cin>>n;
+    lint board[n];  // Length of boards.
+    for(lint i=0; i<n; i++){
+        cin>>board[i]; 
+    }
+    lint k;
+    cin>>k;    // Enter the number of painters.
```
Comment (somya-kapoor) [inline Painter's_Partition_Problem/Painter's_Partition_Problem.cpp]: Add space around >>

File: Painter's_Partition_Problem/Painter's_Partition_Problem.cpp [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+    lint sum = 0;
+    lint m = 0;
+    for(lint i=0; i<n; i++) {
```
Comment (somya-kapoor) [inline Painter's_Partition_Problem/Painter's_Partition_Problem.cpp]: Add space around <,+

File: Painter's_Partition_Problem/Painter's_Partition_Problem.cpp [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+        sum += board[i];
+        m = max(m, board[i]);
+    }
+    cout<<" Minimum time to paint all the boards is : "<<partition(board, k, sum, n, m);
```
Comment (somya-kapoor) [inline Painter's_Partition_Problem/Painter's_Partition_Problem.cpp]: Add space before <<

File: Painter's_Partition_Problem/Painter's_Partition_Problem.java [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+
+    /*
+        This function returns the minimum no. of painters required to
+        paint the board given that a painter can paint consecutive mid
+        units of board at max.
+    */
+    public static int painters(int board[], int n, int mid) {
+        int painter = 0;
+        int time = 0;
+        for(int i=0; i<n; i++) {
```
Comment (somya-kapoor) [inline Painter's_Partition_Problem/Painter's_Partition_Problem.java]: Add space around =,<

File: Painter's_Partition_Problem/Painter's_Partition_Problem.java [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+            else {
+                low = mid+1;
+            }
+        }
+        return low;
+    }
+
+    public static void main(String[] args){
+        int n = 5; // The number of boards.
+        int[] board = {2,5,6,3,4};  // Length of boards.
```
Comment (somya-kapoor) [inline Painter's_Partition_Problem/Painter's_Partition_Problem.java]: Add space after the comma.

File: Painter's_Partition_Problem/Painter's_Partition_Problem.java [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+        int k = 4;   // The number of painters.
+        int sum = 0;
+        int m = 0;
+        for(int i=0; i<n; i++) {
```
Comment (somya-kapoor) [inline Painter's_Partition_Problem/Painter's_Partition_Problem.java]: Add space around =,<

File: Painter's_Partition_Problem/Painter's_Partition_Problem.java [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+            sum += board[i];
+            m = Math.max(m, board[i]);
+        }
+        System.out.print(" Minimum time to paint all the boards is : "+partition(board, k, sum, n, m));
```
Comment (somya-kapoor) [inline Painter's_Partition_Problem/Painter's_Partition_Problem.java]: Remove space before :

**Comment by MastersAbh** [comment]
@somya-kapoor , There is no mention of space around < and = in the coding standards that are being followed in the repository. Please take back your requested changes, because I have followed the required standards.

**Comment by MastersAbh** [comment]
Made the changes. Review @somya-kapoor

**Comment by somya-kapoor** [comment]
@jainaman224 This pr is approved . Please have a look.

**Pull request merged**